File: 102 Binary Tree Level Order Traversal/BinaryTreeLevelOrderTraversal_BFS.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class BinaryTreeLevelOrderTraversal_BFS {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new LinkedList<List<Integer>>();
        if (root == null) { // base case
            return res;
        }
        Queue<TreeNode> queue = new LinkedList<TreeNode>();
        TreeNode curr = root;
        queue.add(curr);
        while (!queue.isEmpty()) {
            List<Integer> level = new LinkedList<>();
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                curr = queue.poll();
                level.add(curr.val);
                if (curr.left != null) {
                    queue.add(curr.left);
                }
                if (curr.right != null) {
                    queue.add(curr.right);
                }
            }
            res.add(level);
        }
        return res;
    }
}

File: 0107 Binary Tree Level Order Traversal II/binary_tree_level_order_traversal_bfs.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        res = []
        curr = root
        if curr == None:
            return res
        q = [curr] # initialize the queue
        while (len(q) != 0):
            level = []
            length = len(q)
            for _ in range(length):
                curr = q.pop(0)
                level.append(curr.val)
                if curr.left != None:
                    q.append(curr.left)
                if curr.right != None:
                    q.append(curr.right)
            res.insert(0, level)
        return res